version: '3.8'

networks:
  planningNet:
    driver: bridge

volumes:
  node_modules:

services:

  lets_planning:
    container_name: lets_planning
    build:
      context: ./app
      dockerfile: Dockerfile
    volumes:
      - ./app:/usr/src/app
      - node_modules:/usr/src/app/node_modules
    networks:
      - planningNet
    tty: true
    ports:
      - "3000:3000"
    depends_on:
      - database

  adminer:
      image: adminer
      container_name: mysql_admin
      restart: always
      ports:
        - 8081:8080
      networks:
      - planningNet
      environment:
        - ADMINER_DEFAULT_SERVER=mysql_database
        - ADMINER_DESIGN=dracula
      depends_on:
        mysql_database:
        condition: service_healthy

  database:
    image: mysql:5.7
    container_name: mysql_database
    command: --innodb-use-native-aio=0
    restart: always
    tty: true
    ports:
      - 3306:3306
    expose:
      - "3306"
    volumes:
      - ./databaseFiles:/var/lib/mysql
    environment:
      - MYSQL_DATABASE=nodedb
      - MYSQL_ROOT_PASSWORD=root      
    networks:
      - planningNet
    healthcheck:
      test: "mysql -uroot -proot --execute \"SHOW DATABASES;\""
      interval: 20s
      timeout: 20s
      retries: 20